internal/fanoutconsumer/metrics.go

```diff
@@ -42,7 +42,7 @@ type metricsConsumer struct {
 }
 
 func (msc *metricsConsumer) Capabilities() consumer.Capabilities {
-	return consumer.Capabilities{MutatesData: false}
+	return consumer.Capabilities{MutatesData: len(msc.mutable) > 0}
```

Comment:
If there is a readonly receiver in the fan-out, it's not mutating. The mutable data will be copied
```suggestion
	return consumer.Capabilities{MutatesData: len(msc.mutable) > 0 && len(msc. readonly) == 0}
```